### PR TITLE
Update Proof Gen

### PR DESCRIPTION
**Pull request by BasileiosKal**
Based on the WG discussions, update the BBS Proof based on [CDL16].

(Uses  #287)

### CODE REVIEW

**Comment by BasileiosKal** [comment]
@christianpaquin, @Wind4Greg ^these are breaking changes for the proof gen.

Have included test vectors with additional traces to hopefully help with updating 🙏

**Comment by christianpaquin** [comment]
> @christianpaquin, @Wind4Greg ^these are breaking changes for the proof gen.
> 
> Have included test vectors with additional traces to hopefully help with updating 🙏

@BasileiosKal, I updated my implementation ([here](https://github.com/microsoft/bbs-node-reference/pull/30)); test vectors pass.

File: draft-irtf-cfrg-bbs-signatures.md [hunk 1]
```diff
@@ -972,33 +980,35 @@ Outputs:
...
 Procedure:
 
-1. domain = calculate_domain(PK, Q_1, (H_1, ..., H_L), header, api_id)
-2. D = P1 + Q_1 * domain + H_i1 * msg_i1 + ... + H_iR * msg_iR
-3. T =  Abar * r2^ + Bbar * r3^ + H_j1 * m^_j1 + ... +  H_jU * m^_jU
-4. T = T + D * c
-5. return (Abar, Bbar, T, domain)
+5. domain = calculate_domain(PK, Q_1, (H_1, ..., H_L), header, api_id)
+
+7. T1 = Bbar * c + Abar * e^ + D * r1^
```
Comment (christianpaquin) [inline draft-irtf-cfrg-bbs-signatures.md:1007]: Step number should be 6

**Review by christianpaquin — APPROVED** [review]

File: draft-irtf-cfrg-bbs-signatures.md (hunk 1 above)
Comment (BasileiosKal) [reply draft-irtf-cfrg-bbs-signatures.md]: Fixed. Thank you!

**Review by Wind4Greg — APPROVED** [review]
All test vectors confirmed against proof generation/verification update to specification via my independent JavaScript based implementation. Enhanced information (PNG scalars) in test vectors are a great help to implementers!

**Comment by BasileiosKal** [comment]
Multiple approvals! Discussed on WG call at the 4th of December. Test Vectors cross validated. Merging.

Thank you everyone for your help!

**Pull request merged**